core/server/worker/src/main/java/alluxio/worker/block/SpaceReserver.java

```diff
@@ -69,19 +71,25 @@ public SpaceReserver(BlockWorker blockWorker) {
       PropertyKey tierReservedSpaceProp =
           PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO.format(ordinal);
       if (Configuration.containsKey(tierReservedSpaceProp)) {
-        LOG.warn("The property reserved.ratio is deprecated use high/low watermark instead.");
+        LOG.warn("The property reserved.ratio is deprecated, use high/low watermark instead.");
         reservedSpace = (long) (tierCapacity * Configuration.getDouble(tierReservedSpaceProp));
       } else {
         // High watermark defines when to start the space reserving process
         PropertyKey tierHighWatermarkProp =
             PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_HIGH_WATERMARK_RATIO.format(ordinal);
+        Preconditions.checkArgument(Configuration.getDouble(tierHighWatermarkProp) > 0,
+            "The high watermark of tier " + ordinal + " should be positive");
+        Preconditions.checkArgument(Configuration.getDouble(tierHighWatermarkProp) < 1,
+            "The high watermark of tier " + ordinal + " should be less than 100%");
         long highWatermark =
             (long) (tierCapacity * Configuration.getDouble(tierHighWatermarkProp));
         mHighWatermarks.put(tierAlias, highWatermark);
 
         // Low watermark defines when to stop the space reserving process if started
         PropertyKey tierLowWatermarkProp =
             PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_LOW_WATERMARK_RATIO.format(ordinal);
+        Preconditions.checkArgument(Configuration.getDouble(tierLowWatermarkProp) > 0,
+            "The low watermark of tier " + ordinal + " should be positive");
```

Comment:
I think its okay for low watermark to be 0?

Reply:
Theoretically yes.